src/course-unit/course-sequence/hooks.js

```diff
@@ -0,0 +1,139 @@
+import { useSelector } from 'react-redux';
+import { useLayoutEffect, useRef, useState } from 'react';
+import { useWindowSize } from '@edx/paragon';
+
+import { useModel } from '../../generic/model-store';
+import { RequestStatus } from '../../data/constants';
+import { getCourseSectionVertical, getSequenceStatus, sequenceIdsSelector } from '../data/selectors';
+
+export function useSequenceNavigationMetadata(currentSequenceId, currentUnitId) {
+  const { SUCCESSFUL } = RequestStatus;
+  const sequenceIds = useSelector(sequenceIdsSelector);
+  const sequenceStatus = useSelector(getSequenceStatus);
+  const { nextUrl, prevUrl } = useSelector(getCourseSectionVertical);
+  const sequence = useModel('sequences', currentSequenceId);
```

Comment:
i meant like useState close to useState etc. So that they do not cross.

```
const [] = useState();
const [] = useState();
const state = useSelector();
const state = useSelector();
const {} = useCustomHook();
const {} = useCustomHook();
```